cycfx2prog-0.47/cycfx2prog.cc

```diff
@@ -712,7 +716,50 @@ int main(int argc,char **arg)
                         ++errors;
                     }
                 }
-		else
+                else if(!strcmp(cmd,"getorionserial"))
+                {
+                        int length=0;
+                        unsigned char buffer[64];
+
+                        // Set to a known value so bad reads stick out
+                        for(int x = 0; x < 64; x++)
+                            buffer[x]=0xAA;
+
+                        length=cycfx2.OrionSerialNumberRead(buffer,64);
+                        if(length > 0)
+                        {
+                            for(int x = 0; x < length; x++)
+                            {
+                                fprintf(stdout,"%c", buffer[x]);
+                            }
+                            fprintf(stdout,"\n");
+                        }
+                        else
+                        {
+                             ++errors;
+                        }
+                }
+                else if(!strcmp(cmd,"setorionserial"))
+                {
+                    // if a serial number string has been provided and no more than 32 characters
+                    if(a[0] && *a[0] && (strlen(a[0]) < 33))
```

Comment:
Not sure why it is 32 characters ? doesn't we write ascii 64 characters directly ?
Or we write binary 32 characters ?

Reply:
Good Catch! Thanks Keith!